### PR TITLE
[MAINTENANCE] Use `Session` for all `api.greatexpectations.io` calls

### PR DESCRIPTION
**Pull request by Kilo59**
Use `http.get_session()` for all `api.greatexpectations.io` calls.

These calls are automatically logged at `INFO` level.

Update `GxCloudError` to accept a `request.Response` so that HTTP errors are easier to debug.

Move some fluent tests and fixtures around.

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted

    ```
    black .

    ruff . --fix
    ```
- [x] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/data_context/data_context/abstract_data_context.py [hunk 1]
```diff
@@ -4645,9 +4646,17 @@ def _init_datasources(self) -> None:
...
+            f"config.fluent_datasources ->\n{pf(config.fluent_datasources, depth=2)}"
+        )
+        logger.warning(f"config.datasources ->\n{pf(config.datasources, depth=2)}")
+
         if self._datasource_store.cloud_mode:
             for fds in config.fluent_datasources.values():
                 self._add_fluent_datasource(**fds)._rebuild_asset_data_connectors()
+            for ds in config.datasources.values():
+                if isinstance(ds, FluentDatasource):
+                    self._add_fluent_datasource(**ds)._rebuild_asset_data_connectors()
```
Comment (Kilo59) [inline great_expectations/data_context/data_context/abstract_data_context.py]: Can revert this, it's not the problem.

File: tests/datasource/fluent/conftest.py
Comment (Kilo59) [inline tests/datasource/fluent/conftest.py:1]: Just moving around tests and fixtures here.
Nothing is new.

File: great_expectations/datasource/fluent/sources.py [hunk 2]
```diff
@@ -560,11 +562,11 @@ def add_or_update_datasource(
...
             )
-            if id_ and name_or_datasource:
-                if isinstance(name_or_datasource, str):
-                    kwargs["id"] = id_
-                else:
+            if id_:
+                if name_or_datasource and not isinstance(name_or_datasource, str):
                     name_or_datasource.id = id_
+                else:
+                    kwargs["id"] = id_
```
Comment (Kilo59) [inline great_expectations/datasource/fluent/sources.py:569]: This was the cause of the `name` positional args working but the keyword arg not working.

File: great_expectations/data_context/store/gx_cloud_store_backend.py [hunk 3]
```diff
@@ -382,17 +392,17 @@ def _set(  # type: ignore[override]
...
             raise StoreBackendError(
                 f"Unable to set object in GX Cloud Store Backend: {get_user_friendly_error_message(http_exc)}"
-            )
+            ) from http_exc
         except requests.Timeout as timeout_exc:
             logger.exception(timeout_exc)
             raise StoreBackendTransientError(
                 "Unable to set object in GX Cloud Store Backend: This is likely a transient error. Please try again."
-            )
+            ) from timeout_exc
```
Comment (Kilo59) [inline great_expectations/data_context/store/gx_cloud_store_backend.py:400]:
```suggestion
            )
```

File: tests/datasource/fluent/test_contexts.py [hunk 4]
```diff
@@ -175,6 +176,31 @@ def test_cloud_add_or_update_datasource_kw_vs_positional(
...
+    )
+
+    # pass name as positional arg
+    datasource3 = empty_cloud_context_fluent.sources.add_or_update_pandas_filesystem(
+        name, base_directory=taxi_data_samples_dir
+    )
+
+    assert datasource1 == datasource2 == datasource3
+
+
```
Comment (Kilo59) [inline tests/datasource/fluent/test_contexts.py:203]:
```suggestion
```

File: tests/datasource/fluent/conftest.py
Comment (Kilo59) [inline tests/datasource/fluent/conftest.py:1]: I'm simply moving these fixtures to the fluent `conftest` so that they can be re-used and we can do more integration tests.

File: tests/datasource/fluent/test_contexts.py [hunk 5]
```diff
@@ -254,5 +255,33 @@ def test_cloud_context_delete_datasource(
     assert response2.status_code == 404
 
 
+def test_data_connectors_are_built_on_config_load(
```
Comment (Kilo59) [inline tests/datasource/fluent/test_contexts.py:258]: moving this test from `test_viral_snippets.py`

File: great_expectations/exceptions/exceptions.py [hunk 6]
```diff
@@ -454,6 +459,12 @@ class GXCloudError(GreatExpectationsError):
     Generic error used to provide additional context around Cloud-specific issues.
     """
 
+    response: requests.Response
+
+    def __init__(self, message, response: requests.Response) -> None:
```
Comment (cdkini) [inline great_expectations/exceptions/exceptions.py]: Could we annotate message with `str`?

**Review by cdkini — APPROVED** [review]

File: tests/datasource/fluent/conftest.py
Comment (Kilo59) [inline tests/datasource/fluent/conftest.py:1]: Just moving fixtures here. Nothing is changing.
This really should be a different PR but ...

File: great_expectations/exceptions/exceptions.py (hunk 6 above)
Comment (Kilo59) [reply great_expectations/exceptions/exceptions.py]: Done
